apisix/core/config_etcd.lua

```diff
@@ -241,7 +241,11 @@ local function sync_data(self)
 
         else
             if not dir_res.nodes then
-                dir_res.nodes = {}
+                if dir_res.key then
+                    dir_res.nodes = { clone_tab(dir_res) }
+                else
+                    dir_res.nodes = {}
+                end
```

Comment:
Should remove the keys: https://github.com/apache/apisix/blob/aff1a907a3671bd6610b304a54ebef47022ca6cf/apisix/cli/etcd.lua#L206
to see if this idea works.